lua/package-info/ui/dependency-name-input.lua

```diff
@@ -0,0 +1,92 @@
+local Input = require("nui.input")
+
+local logger = require("package-info.logger")
+local safe_call = require("package-info.utils.safe-call")
+
+local M = {}
+
+--- Spawn a new input for the package name
+-- @param props.on_submit: function - executed after selection
+-- @param props.on_close?: function - executed if user closes the input
+-- @param props.on_error?: function - executed if users enters an invalid package name
+M.new = function(props)
+    local style = {
+        relative = "cursor",
+        position = {
+            row = 0,
+            col = 0,
+        },
+        size = {
+            width = 50,
+            height = 2,
+        },
+        highlight = "Normal:Normal",
```

Comment:
`border.highlight` will still continute to work, but it's removed from the documentation and is deprecated.
You might want to do this going forward for overriding border highlight group:

```suggestion
        highlight = "Normal:Normal,FloatBorder:Normal",
```

Updated docs for border highlighting is here: https://github.com/MunifTanjim/nui.nvim/blob/f0a375fa/lua/nui/popup/README.md#borderstyle

Reply:
Oh, I remember seeing that in the docs but deleted thinking I don't need it :smile:. Thanks for the heads-up. 

What do you think about accepting a different format like: 

```lua
highlight = {
    "Normal" = "Normal",
    "FloatBorder" = "Normal"
}
```

I think it would be easier to maintain for the users instead of a big string.

Reply:
Ah, thinking about that, `popup_options.highlight` was actually deprecated even earlier than `border.highlight`. 😂 
https://github.com/MunifTanjim/nui.nvim/commit/163d99a3fb9fe9a630c96bbf6159973885c3caf3 about 6 months ago!

And the recommended option to override window highlight is actually `win_options.winhighlight`.

The reason for these deprecations was to get closer to the `:help 'winhighlight'` option. If something is already supported in neovim core, I'm trying to keep that out of `nui.nvim` and encourage to use the built-in feature. For example: after https://github.com/neovim/neovim/pull/16594 and https://github.com/neovim/neovim/pull/16591 is released in neovim `0.7.0` or `0.8.0`, `popup:map` can be deprecated and users will be able to use the built-in feature.

Reply:
> I think it would be easier to maintain for the users instead of a big string.

I think we can document this utility function somewhere in the wiki 🤔 

```lua
local function get_winhighlight(highlight)
  return table.concat(
    vim.tbl_map(function(key)
      return key .. ":" .. highlight[key]
    end, vim.tbl_keys(highlight)),
    ","
  )
end

local winhighlight = get_winhighlight({ Normal = "Normal", FloatBorder = "Normal" })
```

Reply:
That makes sense yeah. 

Maybe when deprecating settings, check if the user set those and log an error stating the deprecation. I don't think I would have noticed this which so it would just die one day when you remove it :smile:

Reply:
> I don't think I would have noticed this which so it would just die one day when you remove it

😂 Don't worry about it. I don't yet have any plan to delete anything. That's why there's no deprecation message and deprecated things from 6 months ago are still working...

If anything is meant to be deleted, there will be warning messages so that users can notify the plugin maintainers about it. Something like "[nui.nvim] <feature> is deprecated, and will be removed after 3 months from now, please notify the plugin owner about this"